RevealJS Update fails - Cannot find module 'gulp-tap'

### ✅ Have you read and understood the above guidelines?

yes

### 📜 What is the name of the script you are using?

RevealJS

### 📂 What was the exact command used to execute the script?

update

### 📝 Provide a clear and concise description of the issue.

Revealjs does not work anymore

### ⚙️ What settings are you using?

- [x] Default Settings
- [ ] Advanced Settings

### 🖥️ Which Linux distribution are you using?

Debian 12

### 🔄 Steps to reproduce the issue.

Just do update.. 

### ❌ Paste the full error output (if available).

```  
  ✔️  Stopped RevealJS
npm warn deprecated sourcemap-codec@1.4.8: Please use @jridgewell/sourcemap-codec instead
npm warn deprecated q@1.5.1: You or someone you depend on is using Q, the JavaScript Promise library that gave JavaScript developers strong feelings about promises. They can almost certainly migrate to the native JavaScript promise now. Thank you literally everyone for joining me in this bet against the odds. Be excellent to each other.
npm warn deprecated
npm warn deprecated (For a CapTP with native promises, see @endo/eventual-send and @endo/captp)

added 737 packages, and audited 738 packages in 33s

84 packages are looking for funding
  run `npm fund` for details

found 0 vulnerabilities
npm notice
npm notice New major version of npm available! 10.9.2 -> 11.2.0
npm notice Changelog: https://github.com/npm/cli/releases/tag/v11.2.0
npm notice To update run: npm install -g npm@11.2.0
npm notice
  ✔️  Updated RevealJS to 5.2.0
  ✔️  Started RevealJS
  ✔️  Cleanup Completed
  ✔️  Update Successful
```  
  
  
  
  
```  
Mar 22 20:24:41 revealjs systemd[1]: Stopping revealjs.service - Reveal.js Service...
Mar 22 20:24:41 revealjs systemd[1]: revealjs.service: Deactivated successfully.
Mar 22 20:24:41 revealjs systemd[1]: Stopped revealjs.service - Reveal.js Service.
Mar 22 20:24:41 revealjs systemd[1]: revealjs.service: Consumed 1min 54.851s CPU time.
Mar 22 20:25:15 revealjs systemd[1]: Started revealjs.service - Reveal.js Service.
Mar 22 20:25:17 revealjs npm[3263]: > reveal.js@5.2.0 start
Mar 22 20:25:17 revealjs npm[3263]: > gulp serve
Mar 22 20:25:26 revealjs npm[3298]: Error: Cannot find module 'gulp-tap'
Mar 22 20:25:26 revealjs npm[3298]: Require stack:
Mar 22 20:25:26 revealjs npm[3298]: - /opt/revealjs/gulpfile.js
Mar 22 20:25:26 revealjs npm[3298]: - /opt/revealjs/node_modules/gulp-cli/lib/shared/require-or-import.js
Mar 22 20:25:26 revealjs npm[3298]: - /opt/revealjs/node_modules/gulp-cli/lib/versioned/^5.0.0/index.js
Mar 22 20:25:26 revealjs npm[3298]: - /opt/revealjs/node_modules/gulp-cli/index.js
Mar 22 20:25:26 revealjs npm[3298]: - /opt/revealjs/node_modules/gulp/bin/gulp.js
Mar 22 20:25:26 revealjs npm[3298]:     at Function._resolveFilename (node:internal/modules/cjs/loader:1225:15)
Mar 22 20:25:26 revealjs npm[3298]:     at Function._load (node:internal/modules/cjs/loader:1055:27)
Mar 22 20:25:26 revealjs npm[3298]:     at TracingChannel.traceSync (node:diagnostics_channel:322:14)
Mar 22 20:25:26 revealjs npm[3298]:     at wrapModuleLoad (node:internal/modules/cjs/loader:220:24)
Mar 22 20:25:26 revealjs npm[3298]:     at Module.require (node:internal/modules/cjs/loader:1311:12)
Mar 22 20:25:26 revealjs npm[3298]:     at require (node:internal/modules/helpers:136:16)
Mar 22 20:25:26 revealjs npm[3298]:     at Object.<anonymous> (/opt/revealjs/gulpfile.js:15:13)
Mar 22 20:25:26 revealjs npm[3298]:     at Module._compile (node:internal/modules/cjs/loader:1554:14)
Mar 22 20:25:26 revealjs npm[3298]:     at Object..js (node:internal/modules/cjs/loader:1706:10)
Mar 22 20:25:26 revealjs npm[3298]:     at Module.load (node:internal/modules/cjs/loader:1289:32) {
Mar 22 20:25:26 revealjs npm[3298]:   code: 'MODULE_NOT_FOUND',
Mar 22 20:25:26 revealjs npm[3298]:   requireStack: [
Mar 22 20:25:26 revealjs npm[3298]:     '/opt/revealjs/gulpfile.js',
Mar 22 20:25:26 revealjs npm[3298]:     '/opt/revealjs/node_modules/gulp-cli/lib/shared/require-or-import.js',
Mar 22 20:25:26 revealjs npm[3298]:     '/opt/revealjs/node_modules/gulp-cli/lib/versioned/^5.0.0/index.js',
Mar 22 20:25:26 revealjs npm[3298]:     '/opt/revealjs/node_modules/gulp-cli/index.js',
Mar 22 20:25:26 revealjs npm[3298]:     '/opt/revealjs/node_modules/gulp/bin/gulp.js'
Mar 22 20:25:26 revealjs npm[3298]:   ]
Mar 22 20:25:26 revealjs npm[3298]: }
Mar 22 20:25:26 revealjs systemd[1]: revealjs.service: Main process exited, code=exited, status=1/FAILURE
Mar 22 20:25:26 revealjs systemd[1]: revealjs.service: Failed with result 'exit-code'.
Mar 22 20:25:26 revealjs systemd[1]: revealjs.service: Consumed 1.431s CPU time.
Mar 22 20:25:27 revealjs systemd[1]: revealjs.service: Scheduled restart job, restart counter is at 1.
Mar 22 20:25:27 revealjs systemd[1]: Stopped revealjs.service - Reveal.js Service.
Mar 22 20:25:27 revealjs systemd[1]: revealjs.service: Consumed 1.431s CPU time.
Mar 22 20:25:27 revealjs systemd[1]: Started revealjs.service - Reveal.js Service.
Mar 22 20:25:27 revealjs npm[3310]: > reveal.js@5.2.0 start
Mar 22 20:25:27 revealjs npm[3310]: > gulp serve
Mar 22 20:25:28 revealjs npm[3322]: Error: Cannot find module 'gulp-tap'
Mar 22 20:25:28 revealjs npm[3322]: Require stack:
Mar 22 20:25:28 revealjs npm[3322]: - /opt/revealjs/gulpfile.js
Mar 22 20:25:28 revealjs npm[3322]: - /opt/revealjs/node_modules/gulp-cli/lib/shared/require-or-import.js
Mar 22 20:25:28 revealjs npm[3322]: - /opt/revealjs/node_modules/gulp-cli/lib/versioned/^5.0.0/index.js
Mar 22 20:25:28 revealjs npm[3322]: - /opt/revealjs/node_modules/gulp-cli/index.js
Mar 22 20:25:28 revealjs npm[3322]: - /opt/revealjs/node_modules/gulp/bin/gulp.js
Mar 22 20:25:28 revealjs npm[3322]:     at Function._resolveFilename (node:internal/modules/cjs/loader:1225:15)
Mar 22 20:25:28 revealjs npm[3322]:     at Function._load (node:internal/modules/cjs/loader:1055:27)
Mar 22 20:25:28 revealjs npm[3322]:     at TracingChannel.traceSync (node:diagnostics_channel:322:14)
Mar 22 20:25:28 revealjs npm[3322]:     at wrapModuleLoad (node:internal/modules/cjs/loader:220:24)
Mar 22 20:25:28 revealjs npm[3322]:     at Module.require (node:internal/modules/cjs/loader:1311:12)
Mar 22 20:25:28 revealjs npm[3322]:     at require (node:internal/modules/helpers:136:16)
Mar 22 20:25:28 revealjs npm[3322]:     at Object.<anonymous> (/opt/revealjs/gulpfile.js:15:13)
Mar 22 20:25:28 revealjs npm[3322]:     at Module._compile (node:internal/modules/cjs/loader:1554:14)
Mar 22 20:25:28 revealjs npm[3322]:     at Object..js (node:internal/modules/cjs/loader:1706:10)
Mar 22 20:25:28 revealjs npm[3322]:     at Module.load (node:internal/modules/cjs/loader:1289:32) {
Mar 22 20:25:28 revealjs npm[3322]:   code: 'MODULE_NOT_FOUND',
Mar 22 20:25:28 revealjs npm[3322]:   requireStack: [
Mar 22 20:25:28 revealjs npm[3322]:     '/opt/revealjs/gulpfile.js',
Mar 22 20:25:28 revealjs npm[3322]:     '/opt/revealjs/node_modules/gulp-cli/lib/shared/require-or-import.js',
Mar 22 20:25:28 revealjs npm[3322]:     '/opt/revealjs/node_modules/gulp-cli/lib/versioned/^5.0.0/index.js',
Mar 22 20:25:28 revealjs npm[3322]:     '/opt/revealjs/node_modules/gulp-cli/index.js',
Mar 22 20:25:28 revealjs npm[3322]:     '/opt/revealjs/node_modules/gulp/bin/gulp.js'
Mar 22 20:25:28 revealjs npm[3322]:   ]
Mar 22 20:25:28 revealjs npm[3322]: }
Mar 22 20:25:28 revealjs systemd[1]: revealjs.service: Main process exited, code=exited, status=1/FAILURE
Mar 22 20:25:28 revealjs systemd[1]: revealjs.service: Failed with result 'exit-code'.
Mar 22 20:25:28 revealjs systemd[1]: revealjs.service: Scheduled restart job, restart counter is at 2.
Mar 22 20:25:28 revealjs systemd[1]: Stopped revealjs.service - Reveal.js Service.
Mar 22 20:25:28 revealjs systemd[1]: Started revealjs.service - Reveal.js Service.
Mar 22 20:25:28 revealjs npm[3333]: > reveal.js@5.2.0 start
Mar 22 20:25:28 revealjs npm[3333]: > gulp serve
Mar 22 20:25:29 revealjs npm[3345]: Error: Cannot find module 'gulp-tap'
Mar 22 20:25:29 revealjs npm[3345]: Require stack:
Mar 22 20:25:29 revealjs npm[3345]: - /opt/revealjs/gulpfile.js
Mar 22 20:25:29 revealjs npm[3345]: - /opt/revealjs/node_modules/gulp-cli/lib/shared/require-or-import.js
Mar 22 20:25:29 revealjs npm[3345]: - /opt/revealjs/node_modules/gulp-cli/lib/versioned/^5.0.0/index.js
Mar 22 20:25:29 revealjs npm[3345]: - /opt/revealjs/node_modules/gulp-cli/index.js
Mar 22 20:25:29 revealjs npm[3345]: - /opt/revealjs/node_modules/gulp/bin/gulp.js
Mar 22 20:25:29 revealjs npm[3345]:     at Function._resolveFilename (node:internal/modules/cjs/loader:1225:15)
Mar 22 20:25:29 revealjs npm[3345]:     at Function._load (node:internal/modules/cjs/loader:1055:27)
Mar 22 20:25:29 revealjs npm[3345]:     at TracingChannel.traceSync (node:diagnostics_channel:322:14)
Mar 22 20:25:29 revealjs npm[3345]:     at wrapModuleLoad (node:internal/modules/cjs/loader:220:24)
Mar 22 20:25:29 revealjs npm[3345]:     at Module.require (node:internal/modules/cjs/loader:1311:12)
Mar 22 20:25:29 revealjs npm[3345]:     at require (node:internal/modules/helpers:136:16)
Mar 22 20:25:29 revealjs npm[3345]:     at Object.<anonymous> (/opt/revealjs/gulpfile.js:15:13)
Mar 22 20:25:29 revealjs npm[3345]:     at Module._compile (node:internal/modules/cjs/loader:1554:14)
Mar 22 20:25:29 revealjs npm[3345]:     at Object..js (node:internal/modules/cjs/loader:1706:10)
Mar 22 20:25:29 revealjs npm[3345]:     at Module.load (node:internal/modules/cjs/loader:1289:32) {
Mar 22 20:25:29 revealjs npm[3345]:   code: 'MODULE_NOT_FOUND',
Mar 22 20:25:29 revealjs npm[3345]:   requireStack: [
Mar 22 20:25:29 revealjs npm[3345]:     '/opt/revealjs/gulpfile.js',
Mar 22 20:25:29 revealjs npm[3345]:     '/opt/revealjs/node_modules/gulp-cli/lib/shared/require-or-import.js',
Mar 22 20:25:29 revealjs npm[3345]:     '/opt/revealjs/node_modules/gulp-cli/lib/versioned/^5.0.0/index.js',
Mar 22 20:25:29 revealjs npm[3345]:     '/opt/revealjs/node_modules/gulp-cli/index.js',
Mar 22 20:25:29 revealjs npm[3345]:     '/opt/revealjs/node_modules/gulp/bin/gulp.js'
Mar 22 20:25:29 revealjs npm[3345]:   ]
Mar 22 20:25:29 revealjs npm[3345]: }
Mar 22 20:25:29 revealjs systemd[1]: revealjs.service: Main process exited, code=exited, status=1/FAILURE
Mar 22 20:25:29 revealjs systemd[1]: revealjs.service: Failed with result 'exit-code'.
Mar 22 20:25:29 revealjs systemd[1]: revealjs.service: Scheduled restart job, restart counter is at 3.
Mar 22 20:25:29 revealjs systemd[1]: Stopped revealjs.service - Reveal.js Service.
Mar 22 20:25:29 revealjs systemd[1]: Started revealjs.service - Reveal.js Service.
Mar 22 20:25:29 revealjs npm[3356]: > reveal.js@5.2.0 start
Mar 22 20:25:29 revealjs npm[3356]: > gulp serve
Mar 22 20:25:30 revealjs npm[3368]: Error: Cannot find module 'gulp-tap'
Mar 22 20:25:30 revealjs npm[3368]: Require stack:
Mar 22 20:25:30 revealjs npm[3368]: - /opt/revealjs/gulpfile.js
Mar 22 20:25:30 revealjs npm[3368]: - /opt/revealjs/node_modules/gulp-cli/lib/shared/require-or-import.js
Mar 22 20:25:30 revealjs npm[3368]: - /opt/revealjs/node_modules/gulp-cli/lib/versioned/^5.0.0/index.js
Mar 22 20:25:30 revealjs npm[3368]: - /opt/revealjs/node_modules/gulp-cli/index.js
Mar 22 20:25:30 revealjs npm[3368]: - /opt/revealjs/node_modules/gulp/bin/gulp.js
Mar 22 20:25:30 revealjs npm[3368]:     at Function._resolveFilename (node:internal/modules/cjs/loader:1225:15)
Mar 22 20:25:30 revealjs npm[3368]:     at Function._load (node:internal/modules/cjs/loader:1055:27)
Mar 22 20:25:30 revealjs npm[3368]:     at TracingChannel.traceSync (node:diagnostics_channel:322:14)
Mar 22 20:25:30 revealjs npm[3368]:     at wrapModuleLoad (node:internal/modules/cjs/loader:220:24)
Mar 22 20:25:30 revealjs npm[3368]:     at Module.require (node:internal/modules/cjs/loader:1311:12)
Mar 22 20:25:30 revealjs npm[3368]:     at require (node:internal/modules/helpers:136:16)
Mar 22 20:25:30 revealjs npm[3368]:     at Object.<anonymous> (/opt/revealjs/gulpfile.js:15:13)
Mar 22 20:25:30 revealjs npm[3368]:     at Module._compile (node:internal/modules/cjs/loader:1554:14)
Mar 22 20:25:30 revealjs npm[3368]:     at Object..js (node:internal/modules/cjs/loader:1706:10)
Mar 22 20:25:30 revealjs npm[3368]:     at Module.load (node:internal/modules/cjs/loader:1289:32) {
Mar 22 20:25:30 revealjs npm[3368]:   code: 'MODULE_NOT_FOUND',
Mar 22 20:25:30 revealjs npm[3368]:   requireStack: [
Mar 22 20:25:30 revealjs npm[3368]:     '/opt/revealjs/gulpfile.js',
Mar 22 20:25:30 revealjs npm[3368]:     '/opt/revealjs/node_modules/gulp-cli/lib/shared/require-or-import.js',
Mar 22 20:25:30 revealjs npm[3368]:     '/opt/revealjs/node_modules/gulp-cli/lib/versioned/^5.0.0/index.js',
Mar 22 20:25:30 revealjs npm[3368]:     '/opt/revealjs/node_modules/gulp-cli/index.js',
Mar 22 20:25:30 revealjs npm[3368]:     '/opt/revealjs/node_modules/gulp/bin/gulp.js'
Mar 22 20:25:30 revealjs npm[3368]:   ]
Mar 22 20:25:30 revealjs npm[3368]: }
Mar 22 20:25:30 revealjs systemd[1]: revealjs.service: Main process exited, code=exited, status=1/FAILURE
Mar 22 20:25:30 revealjs systemd[1]: revealjs.service: Failed with result 'exit-code'.
Mar 22 20:25:30 revealjs systemd[1]: revealjs.service: Scheduled restart job, restart counter is at 4.
Mar 22 20:25:30 revealjs systemd[1]: Stopped revealjs.service - Reveal.js Service.
Mar 22 20:25:31 revealjs systemd[1]: Started revealjs.service - Reveal.js Service.
Mar 22 20:25:31 revealjs npm[3379]: > reveal.js@5.2.0 start
Mar 22 20:25:31 revealjs npm[3379]: > gulp serve
Mar 22 20:25:31 revealjs npm[3391]: Error: Cannot find module 'gulp-tap'
Mar 22 20:25:31 revealjs npm[3391]: Require stack:
Mar 22 20:25:31 revealjs npm[3391]: - /opt/revealjs/gulpfile.js
Mar 22 20:25:31 revealjs npm[3391]: - /opt/revealjs/node_modules/gulp-cli/lib/shared/require-or-import.js
Mar 22 20:25:31 revealjs npm[3391]: - /opt/revealjs/node_modules/gulp-cli/lib/versioned/^5.0.0/index.js
Mar 22 20:25:31 revealjs npm[3391]: - /opt/revealjs/node_modules/gulp-cli/index.js
Mar 22 20:25:31 revealjs npm[3391]: - /opt/revealjs/node_modules/gulp/bin/gulp.js
Mar 22 20:25:31 revealjs npm[3391]:     at Function._resolveFilename (node:internal/modules/cjs/loader:1225:15)
Mar 22 20:25:31 revealjs npm[3391]:     at Function._load (node:internal/modules/cjs/loader:1055:27)
Mar 22 20:25:31 revealjs npm[3391]:     at TracingChannel.traceSync (node:diagnostics_channel:322:14)
Mar 22 20:25:31 revealjs npm[3391]:     at wrapModuleLoad (node:internal/modules/cjs/loader:220:24)
Mar 22 20:25:31 revealjs npm[3391]:     at Module.require (node:internal/modules/cjs/loader:1311:12)
Mar 22 20:25:31 revealjs npm[3391]:     at require (node:internal/modules/helpers:136:16)
Mar 22 20:25:31 revealjs npm[3391]:     at Object.<anonymous> (/opt/revealjs/gulpfile.js:15:13)
Mar 22 20:25:31 revealjs npm[3391]:     at Module._compile (node:internal/modules/cjs/loader:1554:14)
Mar 22 20:25:31 revealjs npm[3391]:     at Object..js (node:internal/modules/cjs/loader:1706:10)
Mar 22 20:25:31 revealjs npm[3391]:     at Module.load (node:internal/modules/cjs/loader:1289:32) {
Mar 22 20:25:31 revealjs npm[3391]:   code: 'MODULE_NOT_FOUND',
Mar 22 20:25:31 revealjs npm[3391]:   requireStack: [
Mar 22 20:25:31 revealjs npm[3391]:     '/opt/revealjs/gulpfile.js',
Mar 22 20:25:31 revealjs npm[3391]:     '/opt/revealjs/node_modules/gulp-cli/lib/shared/require-or-import.js',
Mar 22 20:25:31 revealjs npm[3391]:     '/opt/revealjs/node_modules/gulp-cli/lib/versioned/^5.0.0/index.js',
Mar 22 20:25:31 revealjs npm[3391]:     '/opt/revealjs/node_modules/gulp-cli/index.js',
Mar 22 20:25:31 revealjs npm[3391]:     '/opt/revealjs/node_modules/gulp/bin/gulp.js'
Mar 22 20:25:31 revealjs npm[3391]:   ]
Mar 22 20:25:31 revealjs npm[3391]: }
Mar 22 20:25:31 revealjs systemd[1]: revealjs.service: Main process exited, code=exited, status=1/FAILURE
Mar 22 20:25:31 revealjs systemd[1]: revealjs.service: Failed with result 'exit-code'.
Mar 22 20:25:32 revealjs systemd[1]: revealjs.service: Scheduled restart job, restart counter is at 5.
Mar 22 20:25:32 revealjs systemd[1]: Stopped revealjs.service - Reveal.js Service.
Mar 22 20:25:32 revealjs systemd[1]: Started revealjs.service - Reveal.js Service.
Mar 22 20:25:32 revealjs npm[3402]: > reveal.js@5.2.0 start
Mar 22 20:25:32 revealjs npm[3402]: > gulp serve
Mar 22 20:25:33 revealjs npm[3414]: Error: Cannot find module 'gulp-tap'
Mar 22 20:25:33 revealjs npm[3414]: Require stack:
Mar 22 20:25:33 revealjs npm[3414]: - /opt/revealjs/gulpfile.js
Mar 22 20:25:33 revealjs npm[3414]: - /opt/revealjs/node_modules/gulp-cli/lib/shared/require-or-import.js
Mar 22 20:25:33 revealjs npm[3414]: - /opt/revealjs/node_modules/gulp-cli/lib/versioned/^5.0.0/index.js
Mar 22 20:25:33 revealjs npm[3414]: - /opt/revealjs/node_modules/gulp-cli/index.js
Mar 22 20:25:33 revealjs npm[3414]: - /opt/revealjs/node_modules/gulp/bin/gulp.js
Mar 22 20:25:33 revealjs npm[3414]:     at Function._resolveFilename (node:internal/modules/cjs/loader:1225:15)
Mar 22 20:25:33 revealjs npm[3414]:     at Function._load (node:internal/modules/cjs/loader:1055:27)
Mar 22 20:25:33 revealjs npm[3414]:     at TracingChannel.traceSync (node:diagnostics_channel:322:14)
Mar 22 20:25:33 revealjs npm[3414]:     at wrapModuleLoad (node:internal/modules/cjs/loader:220:24)
Mar 22 20:25:33 revealjs npm[3414]:     at Module.require (node:internal/modules/cjs/loader:1311:12)
Mar 22 20:25:33 revealjs npm[3414]:     at require (node:internal/modules/helpers:136:16)
Mar 22 20:25:33 revealjs npm[3414]:     at Object.<anonymous> (/opt/revealjs/gulpfile.js:15:13)
Mar 22 20:25:33 revealjs npm[3414]:     at Module._compile (node:internal/modules/cjs/loader:1554:14)
Mar 22 20:25:33 revealjs npm[3414]:     at Object..js (node:internal/modules/cjs/loader:1706:10)
Mar 22 20:25:33 revealjs npm[3414]:     at Module.load (node:internal/modules/cjs/loader:1289:32) {
Mar 22 20:25:33 revealjs npm[3414]:   code: 'MODULE_NOT_FOUND',
Mar 22 20:25:33 revealjs npm[3414]:   requireStack: [
Mar 22 20:25:33 revealjs npm[3414]:     '/opt/revealjs/gulpfile.js',
Mar 22 20:25:33 revealjs npm[3414]:     '/opt/revealjs/node_modules/gulp-cli/lib/shared/require-or-import.js',
Mar 22 20:25:33 revealjs npm[3414]:     '/opt/revealjs/node_modules/gulp-cli/lib/versioned/^5.0.0/index.js',
Mar 22 20:25:33 revealjs npm[3414]:     '/opt/revealjs/node_modules/gulp-cli/index.js',
Mar 22 20:25:33 revealjs npm[3414]:     '/opt/revealjs/node_modules/gulp/bin/gulp.js'
Mar 22 20:25:33 revealjs npm[3414]:   ]
Mar 22 20:25:33 revealjs npm[3414]: }
Mar 22 20:25:33 revealjs systemd[1]: revealjs.service: Main process exited, code=exited, status=1/FAILURE
Mar 22 20:25:33 revealjs systemd[1]: revealjs.service: Failed with result 'exit-code'.
Mar 22 20:25:33 revealjs systemd[1]: revealjs.service: Scheduled restart job, restart counter is at 6.
Mar 22 20:25:33 revealjs systemd[1]: Stopped revealjs.service - Reveal.js Service.
Mar 22 20:25:33 revealjs systemd[1]: revealjs.service: Start request repeated too quickly.
Mar 22 20:25:33 revealjs systemd[1]: revealjs.service: Failed with result 'exit-code'.
Mar 22 20:25:33 revealjs systemd[1]: Failed to start revealjs.service - Reveal.js Service.
```  

```
root@revealjs:/opt# npm install -g npm@11.2.0

removed 13 packages, and changed 54 packages in 5s

24 packages are looking for funding
  run `npm fund` for details
root@revealjs:/opt# systemctl start revealjs
root@revealjs:/opt# systemctl status revealjs
× revealjs.service - Reveal.js Service
     Loaded: loaded (/etc/systemd/system/revealjs.service; enabled; preset: enabled)
     Active: failed (Result: exit-code) since Sat 2025-03-22 20:38:36 CET; 2s ago
   Duration: 921ms
    Process: 3677 ExecStart=/usr/bin/npm start (code=exited, status=1/FAILURE)
   Main PID: 3677 (code=exited, status=1/FAILURE)
        CPU: 917ms

Mar 22 20:38:36 revealjs systemd[1]: revealjs.service: Scheduled restart job, restart counter is at 5.
Mar 22 20:38:36 revealjs systemd[1]: Stopped revealjs.service - Reveal.js Service.
Mar 22 20:38:36 revealjs systemd[1]: revealjs.service: Start request repeated too quickly.
Mar 22 20:38:36 revealjs systemd[1]: revealjs.service: Failed with result 'exit-code'.
Mar 22 20:38:36 revealjs systemd[1]: Failed to start revealjs.service - Reveal.js Service.
```


```
Mar 22 20:38:30 revealjs systemd[1]: Started revealjs.service - Reveal.js Service.
Mar 22 20:38:30 revealjs npm[3583]: > reveal.js@5.2.0 start
Mar 22 20:38:30 revealjs npm[3583]: > gulp serve
Mar 22 20:38:31 revealjs npm[3595]: Error: Cannot find module 'gulp-tap'
Mar 22 20:38:31 revealjs npm[3595]: Require stack:
Mar 22 20:38:31 revealjs npm[3595]: - /opt/revealjs/gulpfile.js
Mar 22 20:38:31 revealjs npm[3595]: - /opt/revealjs/node_modules/gulp-cli/lib/shared/require-or-import.js
Mar 22 20:38:31 revealjs npm[3595]: - /opt/revealjs/node_modules/gulp-cli/lib/versioned/^5.0.0/index.js
Mar 22 20:38:31 revealjs npm[3595]: - /opt/revealjs/node_modules/gulp-cli/index.js
Mar 22 20:38:31 revealjs npm[3595]: - /opt/revealjs/node_modules/gulp/bin/gulp.js
Mar 22 20:38:31 revealjs npm[3595]:     at Function._resolveFilename (node:internal/modules/cjs/loader:1225:15)
Mar 22 20:38:31 revealjs npm[3595]:     at Function._load (node:internal/modules/cjs/loader:1055:27)
Mar 22 20:38:31 revealjs npm[3595]:     at TracingChannel.traceSync (node:diagnostics_channel:322:14)
Mar 22 20:38:31 revealjs npm[3595]:     at wrapModuleLoad (node:internal/modules/cjs/loader:220:24)
Mar 22 20:38:31 revealjs npm[3595]:     at Module.require (node:internal/modules/cjs/loader:1311:12)
Mar 22 20:38:31 revealjs npm[3595]:     at require (node:internal/modules/helpers:136:16)
Mar 22 20:38:31 revealjs npm[3595]:     at Object.<anonymous> (/opt/revealjs/gulpfile.js:15:13)
Mar 22 20:38:31 revealjs npm[3595]:     at Module._compile (node:internal/modules/cjs/loader:1554:14)
Mar 22 20:38:31 revealjs npm[3595]:     at Object..js (node:internal/modules/cjs/loader:1706:10)
Mar 22 20:38:31 revealjs npm[3595]:     at Module.load (node:internal/modules/cjs/loader:1289:32) {
Mar 22 20:38:31 revealjs npm[3595]:   code: 'MODULE_NOT_FOUND',
Mar 22 20:38:31 revealjs npm[3595]:   requireStack: [
Mar 22 20:38:31 revealjs npm[3595]:     '/opt/revealjs/gulpfile.js',
Mar 22 20:38:31 revealjs npm[3595]:     '/opt/revealjs/node_modules/gulp-cli/lib/shared/require-or-import.js',
Mar 22 20:38:31 revealjs npm[3595]:     '/opt/revealjs/node_modules/gulp-cli/lib/versioned/^5.0.0/index.js',
Mar 22 20:38:31 revealjs npm[3595]:     '/opt/revealjs/node_modules/gulp-cli/index.js',
Mar 22 20:38:31 revealjs npm[3595]:     '/opt/revealjs/node_modules/gulp/bin/gulp.js'
Mar 22 20:38:31 revealjs npm[3595]:   ]
Mar 22 20:38:31 revealjs npm[3595]: }
Mar 22 20:38:31 revealjs systemd[1]: revealjs.service: Main process exited, code=exited, status=1/FAILURE
Mar 22 20:38:31 revealjs systemd[1]: revealjs.service: Failed with result 'exit-code'.
Mar 22 20:38:31 revealjs systemd[1]: revealjs.service: Scheduled restart job, restart counter is at 1.
Mar 22 20:38:31 revealjs systemd[1]: Stopped revealjs.service - Reveal.js Service.
Mar 22 20:38:31 revealjs systemd[1]: Started revealjs.service - Reveal.js Service.
Mar 22 20:38:31 revealjs npm[3606]: > reveal.js@5.2.0 start
Mar 22 20:38:31 revealjs npm[3606]: > gulp serve
Mar 22 20:38:32 revealjs npm[3618]: Error: Cannot find module 'gulp-tap'
Mar 22 20:38:32 revealjs npm[3618]: Require stack:
Mar 22 20:38:32 revealjs npm[3618]: - /opt/revealjs/gulpfile.js
Mar 22 20:38:32 revealjs npm[3618]: - /opt/revealjs/node_modules/gulp-cli/lib/shared/require-or-import.js
Mar 22 20:38:32 revealjs npm[3618]: - /opt/revealjs/node_modules/gulp-cli/lib/versioned/^5.0.0/index.js
Mar 22 20:38:32 revealjs npm[3618]: - /opt/revealjs/node_modules/gulp-cli/index.js
Mar 22 20:38:32 revealjs npm[3618]: - /opt/revealjs/node_modules/gulp/bin/gulp.js
Mar 22 20:38:32 revealjs npm[3618]:     at Function._resolveFilename (node:internal/modules/cjs/loader:1225:15)
Mar 22 20:38:32 revealjs npm[3618]:     at Function._load (node:internal/modules/cjs/loader:1055:27)
Mar 22 20:38:32 revealjs npm[3618]:     at TracingChannel.traceSync (node:diagnostics_channel:322:14)
Mar 22 20:38:32 revealjs npm[3618]:     at wrapModuleLoad (node:internal/modules/cjs/loader:220:24)
Mar 22 20:38:32 revealjs npm[3618]:     at Module.require (node:internal/modules/cjs/loader:1311:12)
Mar 22 20:38:32 revealjs npm[3618]:     at require (node:internal/modules/helpers:136:16)
Mar 22 20:38:32 revealjs npm[3618]:     at Object.<anonymous> (/opt/revealjs/gulpfile.js:15:13)
Mar 22 20:38:32 revealjs npm[3618]:     at Module._compile (node:internal/modules/cjs/loader:1554:14)
Mar 22 20:38:32 revealjs npm[3618]:     at Object..js (node:internal/modules/cjs/loader:1706:10)
Mar 22 20:38:32 revealjs npm[3618]:     at Module.load (node:internal/modules/cjs/loader:1289:32) {
Mar 22 20:38:32 revealjs npm[3618]:   code: 'MODULE_NOT_FOUND',
Mar 22 20:38:32 revealjs npm[3618]:   requireStack: [
Mar 22 20:38:32 revealjs npm[3618]:     '/opt/revealjs/gulpfile.js',
Mar 22 20:38:32 revealjs npm[3618]:     '/opt/revealjs/node_modules/gulp-cli/lib/shared/require-or-import.js',
Mar 22 20:38:32 revealjs npm[3618]:     '/opt/revealjs/node_modules/gulp-cli/lib/versioned/^5.0.0/index.js',
Mar 22 20:38:32 revealjs npm[3618]:     '/opt/revealjs/node_modules/gulp-cli/index.js',
Mar 22 20:38:32 revealjs npm[3618]:     '/opt/revealjs/node_modules/gulp/bin/gulp.js'
Mar 22 20:38:32 revealjs npm[3618]:   ]
Mar 22 20:38:32 revealjs npm[3618]: }
Mar 22 20:38:32 revealjs systemd[1]: revealjs.service: Main process exited, code=exited, status=1/FAILURE
Mar 22 20:38:32 revealjs systemd[1]: revealjs.service: Failed with result 'exit-code'.
Mar 22 20:38:32 revealjs systemd[1]: revealjs.service: Scheduled restart job, restart counter is at 2.
Mar 22 20:38:32 revealjs systemd[1]: Stopped revealjs.service - Reveal.js Service.
Mar 22 20:38:33 revealjs systemd[1]: Started revealjs.service - Reveal.js Service.
Mar 22 20:38:33 revealjs npm[3629]: > reveal.js@5.2.0 start
Mar 22 20:38:33 revealjs npm[3629]: > gulp serve
Mar 22 20:38:33 revealjs npm[3641]: Error: Cannot find module 'gulp-tap'
Mar 22 20:38:33 revealjs npm[3641]: Require stack:
Mar 22 20:38:33 revealjs npm[3641]: - /opt/revealjs/gulpfile.js
Mar 22 20:38:33 revealjs npm[3641]: - /opt/revealjs/node_modules/gulp-cli/lib/shared/require-or-import.js
Mar 22 20:38:33 revealjs npm[3641]: - /opt/revealjs/node_modules/gulp-cli/lib/versioned/^5.0.0/index.js
Mar 22 20:38:33 revealjs npm[3641]: - /opt/revealjs/node_modules/gulp-cli/index.js
Mar 22 20:38:33 revealjs npm[3641]: - /opt/revealjs/node_modules/gulp/bin/gulp.js
Mar 22 20:38:33 revealjs npm[3641]:     at Function._resolveFilename (node:internal/modules/cjs/loader:1225:15)
Mar 22 20:38:33 revealjs npm[3641]:     at Function._load (node:internal/modules/cjs/loader:1055:27)
Mar 22 20:38:33 revealjs npm[3641]:     at TracingChannel.traceSync (node:diagnostics_channel:322:14)
Mar 22 20:38:33 revealjs npm[3641]:     at wrapModuleLoad (node:internal/modules/cjs/loader:220:24)
Mar 22 20:38:33 revealjs npm[3641]:     at Module.require (node:internal/modules/cjs/loader:1311:12)
Mar 22 20:38:33 revealjs npm[3641]:     at require (node:internal/modules/helpers:136:16)
Mar 22 20:38:33 revealjs npm[3641]:     at Object.<anonymous> (/opt/revealjs/gulpfile.js:15:13)
Mar 22 20:38:33 revealjs npm[3641]:     at Module._compile (node:internal/modules/cjs/loader:1554:14)
Mar 22 20:38:33 revealjs npm[3641]:     at Object..js (node:internal/modules/cjs/loader:1706:10)
Mar 22 20:38:33 revealjs npm[3641]:     at Module.load (node:internal/modules/cjs/loader:1289:32) {
Mar 22 20:38:33 revealjs npm[3641]:   code: 'MODULE_NOT_FOUND',
Mar 22 20:38:33 revealjs npm[3641]:   requireStack: [
Mar 22 20:38:33 revealjs npm[3641]:     '/opt/revealjs/gulpfile.js',
Mar 22 20:38:33 revealjs npm[3641]:     '/opt/revealjs/node_modules/gulp-cli/lib/shared/require-or-import.js',
Mar 22 20:38:33 revealjs npm[3641]:     '/opt/revealjs/node_modules/gulp-cli/lib/versioned/^5.0.0/index.js',
Mar 22 20:38:33 revealjs npm[3641]:     '/opt/revealjs/node_modules/gulp-cli/index.js',
Mar 22 20:38:33 revealjs npm[3641]:     '/opt/revealjs/node_modules/gulp/bin/gulp.js'
Mar 22 20:38:33 revealjs npm[3641]:   ]
Mar 22 20:38:33 revealjs npm[3641]: }
Mar 22 20:38:33 revealjs systemd[1]: revealjs.service: Main process exited, code=exited, status=1/FAILURE
Mar 22 20:38:33 revealjs systemd[1]: revealjs.service: Failed with result 'exit-code'.
Mar 22 20:38:34 revealjs systemd[1]: revealjs.service: Scheduled restart job, restart counter is at 3.
Mar 22 20:38:34 revealjs systemd[1]: Stopped revealjs.service - Reveal.js Service.
Mar 22 20:38:34 revealjs systemd[1]: Started revealjs.service - Reveal.js Service.
Mar 22 20:38:34 revealjs npm[3652]: > reveal.js@5.2.0 start
Mar 22 20:38:34 revealjs npm[3652]: > gulp serve
Mar 22 20:38:35 revealjs npm[3664]: Error: Cannot find module 'gulp-tap'
Mar 22 20:38:35 revealjs npm[3664]: Require stack:
Mar 22 20:38:35 revealjs npm[3664]: - /opt/revealjs/gulpfile.js
Mar 22 20:38:35 revealjs npm[3664]: - /opt/revealjs/node_modules/gulp-cli/lib/shared/require-or-import.js
Mar 22 20:38:35 revealjs npm[3664]: - /opt/revealjs/node_modules/gulp-cli/lib/versioned/^5.0.0/index.js
Mar 22 20:38:35 revealjs npm[3664]: - /opt/revealjs/node_modules/gulp-cli/index.js
Mar 22 20:38:35 revealjs npm[3664]: - /opt/revealjs/node_modules/gulp/bin/gulp.js
Mar 22 20:38:35 revealjs npm[3664]:     at Function._resolveFilename (node:internal/modules/cjs/loader:1225:15)
Mar 22 20:38:35 revealjs npm[3664]:     at Function._load (node:internal/modules/cjs/loader:1055:27)
Mar 22 20:38:35 revealjs npm[3664]:     at TracingChannel.traceSync (node:diagnostics_channel:322:14)
Mar 22 20:38:35 revealjs npm[3664]:     at wrapModuleLoad (node:internal/modules/cjs/loader:220:24)
Mar 22 20:38:35 revealjs npm[3664]:     at Module.require (node:internal/modules/cjs/loader:1311:12)
Mar 22 20:38:35 revealjs npm[3664]:     at require (node:internal/modules/helpers:136:16)
Mar 22 20:38:35 revealjs npm[3664]:     at Object.<anonymous> (/opt/revealjs/gulpfile.js:15:13)
Mar 22 20:38:35 revealjs npm[3664]:     at Module._compile (node:internal/modules/cjs/loader:1554:14)
Mar 22 20:38:35 revealjs npm[3664]:     at Object..js (node:internal/modules/cjs/loader:1706:10)
Mar 22 20:38:35 revealjs npm[3664]:     at Module.load (node:internal/modules/cjs/loader:1289:32) {
Mar 22 20:38:35 revealjs npm[3664]:   code: 'MODULE_NOT_FOUND',
Mar 22 20:38:35 revealjs npm[3664]:   requireStack: [
Mar 22 20:38:35 revealjs npm[3664]:     '/opt/revealjs/gulpfile.js',
Mar 22 20:38:35 revealjs npm[3664]:     '/opt/revealjs/node_modules/gulp-cli/lib/shared/require-or-import.js',
Mar 22 20:38:35 revealjs npm[3664]:     '/opt/revealjs/node_modules/gulp-cli/lib/versioned/^5.0.0/index.js',
Mar 22 20:38:35 revealjs npm[3664]:     '/opt/revealjs/node_modules/gulp-cli/index.js',
Mar 22 20:38:35 revealjs npm[3664]:     '/opt/revealjs/node_modules/gulp/bin/gulp.js'
Mar 22 20:38:35 revealjs npm[3664]:   ]
Mar 22 20:38:35 revealjs npm[3664]: }
Mar 22 20:38:35 revealjs systemd[1]: revealjs.service: Main process exited, code=exited, status=1/FAILURE
Mar 22 20:38:35 revealjs systemd[1]: revealjs.service: Failed with result 'exit-code'.
Mar 22 20:38:35 revealjs systemd[1]: revealjs.service: Scheduled restart job, restart counter is at 4.
Mar 22 20:38:35 revealjs systemd[1]: Stopped revealjs.service - Reveal.js Service.
Mar 22 20:38:35 revealjs systemd[1]: Started revealjs.service - Reveal.js Service.
Mar 22 20:38:35 revealjs npm[3677]: > reveal.js@5.2.0 start
Mar 22 20:38:35 revealjs npm[3677]: > gulp serve
Mar 22 20:38:36 revealjs npm[3689]: Error: Cannot find module 'gulp-tap'
Mar 22 20:38:36 revealjs npm[3689]: Require stack:
Mar 22 20:38:36 revealjs npm[3689]: - /opt/revealjs/gulpfile.js
Mar 22 20:38:36 revealjs npm[3689]: - /opt/revealjs/node_modules/gulp-cli/lib/shared/require-or-import.js
Mar 22 20:38:36 revealjs npm[3689]: - /opt/revealjs/node_modules/gulp-cli/lib/versioned/^5.0.0/index.js
Mar 22 20:38:36 revealjs npm[3689]: - /opt/revealjs/node_modules/gulp-cli/index.js
Mar 22 20:38:36 revealjs npm[3689]: - /opt/revealjs/node_modules/gulp/bin/gulp.js
Mar 22 20:38:36 revealjs npm[3689]:     at Function._resolveFilename (node:internal/modules/cjs/loader:1225:15)
Mar 22 20:38:36 revealjs npm[3689]:     at Function._load (node:internal/modules/cjs/loader:1055:27)
Mar 22 20:38:36 revealjs npm[3689]:     at TracingChannel.traceSync (node:diagnostics_channel:322:14)
Mar 22 20:38:36 revealjs npm[3689]:     at wrapModuleLoad (node:internal/modules/cjs/loader:220:24)
Mar 22 20:38:36 revealjs npm[3689]:     at Module.require (node:internal/modules/cjs/loader:1311:12)
Mar 22 20:38:36 revealjs npm[3689]:     at require (node:internal/modules/helpers:136:16)
Mar 22 20:38:36 revealjs npm[3689]:     at Object.<anonymous> (/opt/revealjs/gulpfile.js:15:13)
Mar 22 20:38:36 revealjs npm[3689]:     at Module._compile (node:internal/modules/cjs/loader:1554:14)
Mar 22 20:38:36 revealjs npm[3689]:     at Object..js (node:internal/modules/cjs/loader:1706:10)
Mar 22 20:38:36 revealjs npm[3689]:     at Module.load (node:internal/modules/cjs/loader:1289:32) {
Mar 22 20:38:36 revealjs npm[3689]:   code: 'MODULE_NOT_FOUND',
Mar 22 20:38:36 revealjs npm[3689]:   requireStack: [
Mar 22 20:38:36 revealjs npm[3689]:     '/opt/revealjs/gulpfile.js',
Mar 22 20:38:36 revealjs npm[3689]:     '/opt/revealjs/node_modules/gulp-cli/lib/shared/require-or-import.js',
Mar 22 20:38:36 revealjs npm[3689]:     '/opt/revealjs/node_modules/gulp-cli/lib/versioned/^5.0.0/index.js',
Mar 22 20:38:36 revealjs npm[3689]:     '/opt/revealjs/node_modules/gulp-cli/index.js',
Mar 22 20:38:36 revealjs npm[3689]:     '/opt/revealjs/node_modules/gulp/bin/gulp.js'
Mar 22 20:38:36 revealjs npm[3689]:   ]
Mar 22 20:38:36 revealjs npm[3689]: }
Mar 22 20:38:36 revealjs systemd[1]: revealjs.service: Main process exited, code=exited, status=1/FAILURE
Mar 22 20:38:36 revealjs systemd[1]: revealjs.service: Failed with result 'exit-code'.
Mar 22 20:38:36 revealjs systemd[1]: revealjs.service: Scheduled restart job, restart counter is at 5.
Mar 22 20:38:36 revealjs systemd[1]: Stopped revealjs.service - Reveal.js Service.
Mar 22 20:38:36 revealjs systemd[1]: revealjs.service: Start request repeated too quickly.
Mar 22 20:38:36 revealjs systemd[1]: revealjs.service: Failed with result 'exit-code'.
Mar 22 20:38:36 revealjs systemd[1]: Failed to start revealjs.service - Reveal.js Service.
```



```  
root@revealjs:/opt# ls -ahl
total 48K
drwxr-xr-x  5 root root 4.0K Mar 22 20:24 .
drwxr-xr-x 18 root root 4.0K Mar 15 00:08 ..
-rw-r--r--  1 root root 9.3K Mar 22 20:24 gulpfile.js
-rw-r--r--  1 root root 8.4K Mar 22 20:24 index.html
drwxrwxr-x 11 root root 4.0K Mar 22 20:15 revealjs
drwxrwxr-x  9 root root 4.0K Mar 19 12:03 reveal.js-5.2.0
-rw-r--r--  1 root root    6 Mar 22 20:25 RevealJS_version.txt
```  


### 🖼️ Additional context (optional).

RevealJS-Changelog mentions gup-update: https://github.com/hakimel/reveal.js/releases

@tremor021

--- a/ct/revealjs.sh
+++ b/ct/revealjs.sh
@@ -40,12 +40,11 @@ function update_script() {
         tar zxf $temp_file
         rm -rf /opt/revealjs/node_modules/*
         cp /opt/revealjs/index.html  /opt
-        cp /opt/revealjs/gulpfile.js /opt
         cp -rf reveal.js-${RELEASE}/* /opt/revealjs
         cd /opt/revealjs
         $STD npm install
         cp -f /opt/index.html /opt/revealjs
-        cp -f /opt/gulpfile.js /opt/revealjs
+        sed -i '25s/localhost/0.0.0.0/g' /opt/revealjs/gulpfile.js
         echo "${RELEASE}" >/opt/${APP}_version.txt
         msg_ok "Updated $APP to ${RELEASE}"
 

